aws/scenarios/vm/ec2VM/ec2VM.go

```diff
@@ -18,75 +14,42 @@ type Ec2VM struct {
 
 // NewEc2VM creates a new EC2 instance. By default use WithOS(os.UbuntuOS, os.AMD64Arch).
 func NewEc2VM(ctx *pulumi.Context, options ...func(*Params) error) (*Ec2VM, error) {
-	e, err := aws.AWSEnvironment(ctx)
+	env, err := aws.AWSEnvironment(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	params, err := newParams(e, options...)
+	params, err := newParams(env, options...)
 	if err != nil {
 		return nil, err
 	}
 
-	resourceName := params.name
-	if resourceName == "" {
-		resourceName = e.CommonNamer.ResourceName("ec2-instance")
-	}
+	os := params.common.OS
 	instance, err := awsEc2.NewEC2Instance(
-		e,
-		resourceName,
-		params.ami,
-		params.os.GetAMIArch(params.arch),
-		params.instanceType,
+		env,
+		env.CommonNamer.ResourceName(params.common.GetInstanceNameOrDefault("ec2-instance")),
```

Comment:
Parameter should be mandatory, otherwise we cannot have 2 VMs.

Reply:
I assume the common case is to create a single VM and creating more than one VM will not be common. In this case, I think it is better to provide a default name and raises an explicit error message when creating several unnamed VM with a clear explanation about what to do. 
Unfortunately, this has other implication as for example the `AWSEnvironment` has to be uniq  (https://github.com/DataDog/test-infra-definitions/pull/25/commits/1d1628008215700cac50ebac0bd0653261be39bc). I will handle this change in another PR.
Tracked by AETT-55.